src/main/java/dansplugins/factionsystem/commands/AllyCommand.java

```diff
@@ -85,25 +89,30 @@ public void execute(Player player, String[] args, String key) {
 
         messageFaction(
                 faction,
-                translate("&a" + getText("AlertAttemptedAlliance", faction.getName(), otherFaction.getName()))
+                translate("&a" + getText("AlertAttemptedAlliance", faction.getName(), otherFaction.getName())),
+                Objects.requireNonNull(messageService.getLanguage().getString("AlertAttemptedAlliance"))
+                        .replaceAll("#faction_a#", faction.getName())
+                        .replaceAll("#faction_b#", otherFaction.getName())
         );
 
         messageFaction(
                 otherFaction,
-                translate("&a" + getText("AlertAttemptedAlliance", faction.getName(), otherFaction.getName()))
+                translate("&a" + getText("AlertAttemptedAlliance", faction.getName(), otherFaction.getName())),
+                Objects.requireNonNull(messageService.getLanguage().getString("AlertAttemptedAlliance"))
+                        .replaceAll("#faction_a#", faction.getName())
+                        .replaceAll("#faction_b#", otherFaction.getName())
         );
 
         // check if both factions are have requested an alliance
         if (faction.isRequestedAlly(otherFaction.getName()) && otherFaction.isRequestedAlly(faction.getName())) {
             // ally them
             faction.addAlly(otherFaction.getName());
             otherFaction.addAlly(faction.getName());
-
             // message player's faction
-            messageFaction(faction, translate("&a" + getText("AlertNowAlliedWith", otherFaction.getName())));
+            messageFaction(faction, translate("&a" + getText("AlertNowAlliedWith", otherFaction.getName())), Objects.requireNonNull(messageService.getLanguage().getString("AlertNowAlliedWith")).replaceAll("#faction#", otherFaction.getName()));
```

Comment:
use `replace` over `replaceAll` (`replaceAll` is for regex)

Reply:
I use replaceAll because that can use number but Replace can't, however replaceAll still is better. This cannot be forced

Reply:
Try using replace and seeing if the behavior of the program changes.

Reply:
Any instances of .replaceAll have been replaced with .replace.